armory/scenarios/scenario.py

```diff
@@ -200,16 +201,36 @@ def load_attack(self):
 
     def load_dataset(self, eval_split_default="test"):
         dataset_config = self.config["dataset"]
-        eval_split = dataset_config.get("eval_split", eval_split_default)
-        # Evaluate the ART model on benign test examples
-        log.info(f"Loading test dataset {dataset_config['name']}...")
+        name = dataset_config.get("name")
+        log.info(f"Loading test dataset {name}...")
+        if ":" in name:
+            name, version = name.split(":")
+        else:
+            version = None
+
+        batch_size = dataset_config.get("batch_size", 1)
+        split = dataset_config.get("eval_split", eval_split_default)
+        framework = dataset_config.get("framework", "numpy")
+
+        preprocessor_name = dataset_config.get("preprocessor")
+        preprocessor_kwargs = dataset_config.get("preprocessor_kwargs")
+
+        index = dataset_config.get("index")
+        class_ids = dataset_config.get("class_ids")
+
         self.test_dataset = config_loading.load_dataset(
-            dataset_config,
-            epochs=1,
-            split=eval_split,
+            name,
+            version=version,
+            batch_size=batch_size,
             num_batches=self.num_eval_batches,
-            check_run=self.check_run,
+            epochs=1,
+            split=split,
+            framework=framework,
+            preprocessor_name=preprocessor_name,
+            preprocessor_kwargs=preprocessor_kwargs,
             shuffle_files=False,
+            index=index,
+            class_ids=class_ids,
```

Comment:
Since we're already modifying the scenario, let's update the config structure as well. I recommend having mirrored dataset fields at the top level, `"test_dataset"` and `"train_dataset"`. Then we could remove `"eval_split"` and `"train_split"`, and just have `"split"` in each of them.

Then this whole section could be simplified to something like:
```python
dataset_config = copy.deepcopy(self.config["test_dataset"])
name = dataset_config.get("name")
log.info(f"Loading test dataset {name}...")
if dataset_config.get("epochs") != 1:
    raise ValueError("epochs must be 1 for test dataset")
if "shuffle_files" not in dataset_config:
    dataset_config["shuffle_files"] = False
dataset_config["split"] = dataset_config.get("split", eval_split_default)
self.test_dataset = config_load.load_dataset(**dataset_config)
```
and similarly for the train loader.

Reply:
I had a similar thought, of renaming `eval_split` to `split` so this could be done. I had been thinking of moving train split to the `"fit_kwargs"` field, but I didn't like having a dataset param under the `"model"` config. I also don't love changing the top-level config structure but I do think in theory it could make sense here, so perhaps it's worth it

An alternative would be to keep the field names and pop them, but that gets closer to what we're already doing currently
```
datset_config["split"] = datset_config.pop("eval_split", None)
```

I think one minor issue with the snippet you posted is that it doesn't account for `version` or `num_batches=self.num_eval_batches`

Reply:
I think for version, we just make people specify the version explicitly as a key-value pair, and not have it as part of name. For num_eval_batches, you're right.

Reply:
Yeah, I don't like having a dataset param under the model config. Definitely gets messy that way.

Reply:
added a new commit to address this